source/entities/Creature.cpp

```diff
@@ -155,6 +160,8 @@ Creature::Creature(GameMap* gameMap, const CreatureDefinition* definition) :
 
     if(mDefinition->getWeaponSpawnR().compare("none") != 0)
         mWeaponR = gameMap->getWeapon(mDefinition->getWeaponSpawnR());
+
+    setupDefinition(*gameMap, *ConfigManager::getSingleton().getCreatureDefinitionDefaultWorker());
```

Comment:
Ideally it would be always done by the constructor, but I couldn't get that to work for the map loading when I tried.